flocx_market/db/api.py

```diff
@@ -0,0 +1,82 @@
+import pymysql
+from sqlalchemy import create_engine
+from sqlalchemy.orm import sessionmaker
+from flocx_market.db import models as db_models
+from sqlalchemy.sql import text
+
+_ENGINE = None
+_SESSION_MAKER = None
+
+#1.connecting
+def get_engine():
+    global _ENGINE
+    if _ENGINE is not None:
+        return _ENGINE
+
+    _ENGINE = create_engine('mysql+pymysql://flocx_market:qwerty123@127.0.0.1:3306/flocx_market', echo=True)
+    # db_models.Base.metadata.create_all(_ENGINE)
+    return _ENGINE
+
+#2.creating a session
+def get_session_maker(engine):
+    global _SESSION_MAKER
+    if _SESSION_MAKER is not None:
+        return _SESSION_MAKER
+
+    _SESSION_MAKER = sessionmaker(bind=engine)
+    return _SESSION_MAKER
+
+def get_session():
+    engine = get_engine()
+    Session = get_session_maker(engine)
+    session = Session()
+    create_sample_data(engine)
+    return session
+
+def create_sample_data(engine):
```

Comment:
I think I haven't expressed myself well about this.  I apologize for that. Let me try aqgain.

The reason we use something like `sqlalchemy` is so that we don't have to write raw SQL command ourselves.  Rather than passing a SQL statement to the `execute` method, you should be instantiating a new offer like this:

```
import flocx_market.db.models as db_models
import flocx_market.db.api as db_api

import datetime
import uuid

session = db_api.get_session()
offer = db_models.Offer(
    server_name='example',
    server_config={
        'cpu_arch': 'x86_64',
        'cpus': 32,
        'memory_mb': 65536,
    },
    cost=0.0,
    provider_id=str(uuid.uuid4()),
    creator_id=str(uuid.uuid4()),
    marketplace_offer_id=str(uuid.uuid4()),
    marketplace_date_created=datetime.datetime.now(),
    start_time=datetime.datetime.now(),
    end_time=datetime.datetime.now() + datetime.timedelta(days=7),
    duration=24,
)
session.add(offer)
session.commit()
```

That will create an offer and add it to the database.

This code should not be in `flocx_market/db/api.py`; it should probably be a standalone script.